### PR TITLE
Refactor manager and options to use controller-runtime structs

### PR DESCRIPTION
**Pull request by srm09**
**What this PR does / why we need it**:
This patch removes duplicates code and reuses types from controller-runtime

**Which issue(s) this PR fixes**:
Fixes n/a

**Special notes for your reviewer**:
n/a 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by srm09** [comment]
@yastij @gab-satchi 
Check out [this commit](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1143/commits/d4199b6ac252f1661f034567b954856d28e9d553) about the refactoring I was talking about.

**Comment by srm09** [comment]
/retitle Refactor manager and options to use controller-runtime structs

**Comment by yastij** [comment]
it seems like `capv-controller-manager` is crashing from the e2e logs

**Comment by srm09** [comment]
Planning to debug the failure today, by running the tests locally.

**Comment by gab-satchi** [comment]
changes lgtm, just need a squash

File: pkg/manager/manager.go [hunk 1]
```diff
@@ -60,19 +60,7 @@ func New(opts Options) (Manager, error) {
...
 
 	// Build the controller manager.
-	mgr, err := ctrl.NewManager(opts.KubeConfig, ctrl.Options{
-		Scheme:                  opts.Scheme,
-		MetricsBindAddress:      opts.MetricsAddr,
-		LeaderElection:          opts.LeaderElectionEnabled,
-		LeaderElectionID:        opts.LeaderElectionID,
-		LeaderElectionNamespace: opts.LeaderElectionNamespace,
-		SyncPeriod:              &opts.SyncPeriod,
-		Namespace:               opts.WatchNamespace,
```
Comment (gab-satchi) [inline pkg/manager/manager.go:70]: doesn't look like WatchNamespace is plumbed through to opts.Namespace anymore

Comment (srm09) [reply pkg/manager/manager.go:70]: nice catch. :)

**Comment by srm09** [comment]
@gab-satchi PTAL

**Review by gab-satchi — APPROVED** [review]
/lgtm

**Review by yastij — APPROVED** [review]
/approve

**Pull request merged**